runtime/tests/evm_fee_preferences.rs

```diff
@@ -0,0 +1,542 @@
+            );
+            assert_eq!(
+                <GenericAsset as MultiCurrency>::free_balance(&cennznet_address, CENNZ_ASSET_ID),
+                cennz_balance_before
+            );
+        });
+}
+
+#[test]
+fn evm_call_with_fee_preferences_no_balance_should_fail() {
+	let eth_address: H160 = hex!("420aC537F1a4f78d4Dfb3A71e902be0E3d480AFB").into();
+	let cennznet_address: AccountId = PrefixedAddressMapping::into_account_id(eth_address);
+	let initial_balance = 1000 * DOLLARS;
+	let initial_liquidity = 500 * DOLLARS;
+
+	ExtBuilder::default()
+        .initial_balance(initial_balance)
+        .stash(initial_balance)
+        .build()
+        .execute_with(|| {
+            // Alice sets up CENNZ <> CPAY liquidity
+            assert_ok!(Cennzx::add_liquidity(
+				Origin::signed(alice()),
+				CENNZ_ASSET_ID,
+				initial_liquidity, // min. liquidity
+				initial_liquidity, // liquidity CENNZ
+				initial_liquidity, // liquidity CPAY
+			));
+
+            let cpay_balance_before = <GenericAsset as MultiCurrency>::free_balance(&cennznet_address, CPAY_ASSET_ID);
+            let cennz_balance_before = <GenericAsset as MultiCurrency>::free_balance(&cennznet_address, CENNZ_ASSET_ID);
+
+            // Create input parameters for call
+            let abi = hex!("ccf39ea90000000000000000000000000000000000000000000000000000000000003e8000000000000000000000000000000000000000000000000000000000000000320000000000000000000000001122334455667788991122334455667788990000000000000000000000000000000000000000000000000000000000000000008000000000000000000000000000000000000000000000000000000000000000051234567890000000000000000000000000000000000000000000000000000000");
+            let input = abi.to_vec();
+            let max_fee_per_gas = U256::from(20000000000000u64);
+            let max_priority_fee_per_gas = U256::from(1000000u64);
+            let access_list: Vec<(H160, Vec<H256>)> = vec![];
+            let config: EvmConfig = CENNZNET_EVM_CONFIG.clone();
+            assert_eq!(
+                <Runtime as pallet_evm::Config>::Runner::call(
+                    eth_address,
+                    H160::from_low_u64_be(FEE_PROXY),
+                    input,
+                    U256::from(0u64),
+                    100000u64,
+                    Some(max_fee_per_gas),
+                    Some(max_priority_fee_per_gas),
+                    None,
+                    access_list,
+                    &config
+                )
+                    .is_err(),
+                true
+            );
+
+            // CPAY and CENNZ balance should be unchanged as the transaction never went through
+            assert_eq!(
+                cpay_balance_before,
+                <GenericAsset as MultiCurrency>::free_balance(&cennznet_address, CPAY_ASSET_ID)
+            );
+            assert_eq!(
+                <GenericAsset as MultiCurrency>::free_balance(&cennznet_address, CENNZ_ASSET_ID),
+                cennz_balance_before
+            );
```

Comment:
it's not clear if this should be 0 or not, would do `assert!(balance.is_zero())`